homeassistant/components/broadlink/climate.py

```diff
@@ -0,0 +1,125 @@
+"""Support for Broadlink climate devices."""
+
+import voluptuous as vol
+
+from homeassistant.components.climate import PLATFORM_SCHEMA, ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_IDLE,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, CONF_HOST, TEMP_CELSIUS
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+
+from .const import DOMAIN
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {vol.Required(CONF_HOST): cv.string}, extra=vol.ALLOW_EXTRA
+)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Broadlink climate entities."""
+    device = hass.data[DOMAIN].devices[config_entry.entry_id]
+
+    if device.api.type in {"Hysen heating controller"}:
+        climate_entities = [BroadlinkHysen(device)]
+    async_add_entities(climate_entities)
+
+
+class BroadlinkHysen(ClimateEntity):
+    """Representation of a Broadlink Hysen climate entity."""
+
+    def __init__(self, device):
+        """Initialize the climate entity."""
+        self._device = device
+        self._coordinator = device.update_manager.coordinator
+        self._supported_features = SUPPORT_TARGET_TEMPERATURE
+        self._name = "Broadlink Thermosthat"
+
+    @property
+    def name(self):
+        """Return the name of the switch."""
+        return self._name
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return self._supported_features
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        self._coordinator.data["thermostat_temp"] = temperature
```

Comment:
It's a bit weird that we change the coordinator data from here.

Reply:
Yes it's wierd but required to get the device stable. I'am not entirely sure why but it has something todo with the async requests. If I'am not setting it there sometimes the set_temperature is getting overwritten by the old value while scrolling the temperature wheel in the GUI.